engine/src/main/java/org/terasology/network/internal/NetworkSystemImpl.java

```diff
@@ -797,8 +802,34 @@ private void processRemovedClient(Client client) {
 
     private void processNewClient(NetClient client) {
         client.connected(entityManager, entitySerializer, eventSerializer, eventLibrary);
-        // log after connect so that the name has been set:
-        logger.info("New client connected: {}", client.getName());
+        ServerConnectListManager serverConnectListManager = ServerConnectListManager.getInstance();
+        serverConnectListManager.loadLists();
```

Comment:
This section from line 806 through 831 is duplicating code and functionality with ServerConnectListManager.   Let ServerConnectListManager do the set up and checking for validity.

Something along the lines of the following should be sufficient for NetworkSystemImpl.
`
if (!serverConnectListManager.isClientAllowedToConnect(client)) {
    forceDisconnect(client);
    return;
}
`
By doing it this way, you encapsulate the validity of a client connection outside of this class.   Later on (probably sooner rather than later), we might want to allow access based on ip address in addition to client id.   Or by a public/private key.   Having all these checks in a single class makes it easy to extend and makes this class easier to read.

Reply:
I just did this locally and should commit it later today.